modules/unsupported/geojson-core/src/main/java/org/geotools/data/geojson/GeoJSONReader.java

```diff
@@ -436,6 +451,24 @@ private SimpleFeature getNextFeature(ObjectNode node) throws IOException {
         return feature;
     }
 
+    private String getId(ObjectNode node) {
+        String newId = null;
+        if (node.has(ID)) {
+            if (idPrefix != null) {
+                newId = idPrefix + "." + node.get(ID).asText();
+            } else {
+                newId = node.get(ID).asText();
```

Comment:
I think this should still have the `basename` prefixing the `ID` attribute value

Reply:
I'd argue a low-level GeoJSONReader should be truthful to the GeoJSON content, and be allowed to return exactly the id found in the document. Having an option to use the prefix, for datastore usage, allows for compatibilty.

Reply:
OK, that makes sense